### DISCUSSION TITLE
[BUG] unable to salvage volume pvc-058a766b-94c7-47e2-8e24-72321aabd6f4: Disk with UUID ...on node ... is unschedulable for replica ...

### DISCUSSION THREAD

**Discussion by fzyzcjy** [longhorn/longhorn#4799] [General]
## Describe the bug

When I want to salvage, it reports:

> unable to salvage volume pvc-058a766b-94c7-47e2-8e24-72321aabd6f4: Disk with UUID ab75cc7a-a119-46bc-8897-096dc20cf7f7 on node ecs-132723 is unschedulable for replica pvc-058a766b-94c7-47e2-8e24-72321aabd6f4-r-3ef78437

## To Reproduce

1. Go to longhorn ui
2. click salvage for a failed volume

## Expected behavior

do salvage

## Log or Support bundle

If applicable, add the Longhorn managers' log or support bundle when the issue happens. 
You can generate a Support Bundle using the link at the footer of the Longhorn UI.

## Environment

 - Longhorn version: v1.2.4
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl):  helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster:1
 - Node config
   - OS type and version:linux
   - CPU per node:
   - Memory per node:
   - Disk type(e.g. SSD/NVMe):
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
 - Number of Longhorn volumes in the cluster:

**Comment by derekbit** [comment]
Can you check the longhorn-manager log to see why the replica failed to schedule?

**Comment by fzyzcjy** [comment]
@derekbit No logs indeed?

![image](https://user-images.githubusercontent.com/5236035/198867917-e8a42b3b-af69-46cf-b65a-3925edf0dfa1.png)

**Comment by derekbit** [comment]
Can you check the conditions of the node disks by `kubectl -n longhorn-system get nodes.longhorn.io <node-name> -o yaml` or `kubectl -n longhorn-system get nodes.longhorn.io -o yaml`?

**Comment by fzyzcjy** [comment]
```yaml
apiVersion: longhorn.io/v1beta1
kind: Node
metadata:
  creationTimestamp: '2022-06-27T02:02:14Z'
  finalizers:
    - longhorn.io
  generation: 1
  managedFields:
    - apiVersion: longhorn.io/v1beta1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:finalizers:
            .: {}
            v:"longhorn.io": {}
        f:spec: {}
        f:status: {}
      manager: longhorn-manager
      operation: Update
      time: '2022-06-27T02:02:16Z'
  name: ecs-132723
  namespace: common
  resourceVersion: '31211165'
  selfLink: /apis/longhorn.io/v1beta1/namespaces/common/nodes/ecs-132723
  uid: 26a...
status:
  conditions:
    MountPropagation:
      lastProbeTime: ''
      lastTransitionTime: '2022-06-27T02:02:15Z'
      message: ''
      reason: ''
      status: 'True'
      type: MountPropagation
    Ready:
      lastProbeTime: ''
      lastTransitionTime: '2022-10-29T02:36:06Z'
      message: Node ecs-... is ready
      reason: ''
      status: 'True'
      type: Ready
    Schedulable:
      lastProbeTime: ''
      lastTransitionTime: '2022-06-27T02:02:14Z'
      message: ''
      reason: ''
      status: 'True'
      type: Schedulable
  diskStatus:
    default-disk-f00...90:
      conditions:
        Ready:
          lastProbeTime: ''
          lastTransitionTime: '2022-08-28T19:32:35Z'
          message: >-
            Disk default-disk-f00...90(/mnt/data/tom-longhorn) on node
            ecs-... is ready
          reason: ''
          status: 'True'
          type: Ready
        Schedulable:
          lastProbeTime: ''
          lastTransitionTime: '2022-09-18T01:05:51Z'
          message: >-
            the disk default-disk-f0...90(/mnt/data/tom-longhorn) on
            the node ecs-... has 29884416000 available, but requires reserved
            25204012646, minimal 25% to schedule more replicas
          reason: DiskPressure
          status: 'False'
          type: Schedulable
      diskUUID: ab7..7
      scheduledReplica:
        pvc-xxxxx: 21474836480
        ...
      storageAvailable: 29884416000
      storageMaximum: 136857833472
      storageScheduled: 93520396288
  region: ''
  zone: ''
spec:
  allowScheduling: true
  disks:
    default-disk-f0..90:
      allowScheduling: true
      evictionRequested: false
      path: /mnt/data/tom-longhorn
      storageReserved: 25204012646
      tags: null
  engineManagerCPURequest: 0
  evictionRequested: false
  name: ecs-132723
  replicaManagerCPURequest: 0
  tags: []

**Comment by fzyzcjy** [comment]
I see the message. Looks like b/c the 25% requirement. Thank you very much!

**Comment by fzyzcjy** [comment]
Btw a bit confused about the detail numbers: 29884416000 available, require reserve 25204012646. so should be able to reserve it

**Comment by derekbit** [comment]
The storageMaximum is 136857833472 and `Storage Minimal Available Percentage` is 25%. If the storageAvailable is smaller than 136857833472*25%, the replica scheduling will fail.

ref: https://longhorn.io/docs/1.3.2/references/settings/#storage-minimal-available-percentage

**Comment by fzyzcjy** [comment]
Thank you!
